homeassistant/components/switch/amcrest.py

```diff
@@ -38,11 +38,11 @@ def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
 class AmcrestSwitch(ToggleEntity):
     """Representation of an Amcrest IP camera switch."""
 
-    def __init__(self, setting, camera):
+    def __init__(self, setting, camera, name):
         """Initialize the Amcrest switch."""
         self._setting = setting
         self._camera = camera
-        self._name = SWITCHES[setting][0]
```

Comment:
Should this be a breaking change or possibly a configuration option? There is no unique id for these entities so users would get other names after upgrade.

Reply:
Odd also since I changed this line as well

Reply:
You need to decide how to handle this change. At the least you need to mark it as a breaking change

Reply:
It is already changed. Can you look at my pull request? What you are showing is my pre change code.

Reply:
The passing of the name is correct now but you still need to consider your change of the name. But it is still a breaking change so you need to add a section to the PR about the change.

Reply:
I don't understand why you are not looking at the entire pull. You have only looked at a single change. Sorry I submitted them in bits and pieces but if you look at the pull request, these are the very lines of code I am proposing change for. As you should be able to see from my original pull:


    def __init__(self, setting, camera, name):
        """Initialize the Amcrest switch."""
        self._setting = setting
        self._camera = camera
        self._name = '{} {}'.format(SWITCHES[setting][0], name)
        self._icon = SWITCHES[setting][1]
        self._state = None